src/edu/washington/escience/myria/CsvTupleWriter.java

```diff
@@ -79,4 +95,14 @@ public void error() throws IOException {
       csvPrinter.close();
     }
   }
+
+  private void writeBBtoFile(final ByteBuffer bb, final String fn) throws IOException {
+    boolean append = false;
```

Comment:
Several variables can be inlined, e.g. just put `false` instead of `append` in `FileOutpuyStream()`. Also I think closing the `FileChannel` is enough, and you can work with a `FileChannel` wihout declaring a `FileOutputStream`.